torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -811,6 +824,20 @@ def _post_backward_hook(
                 # Delay using sharded gradient views until after the
                 # reduce-scatter instead of immediately after resharding
                 handle._use_sharded_grad_views()
+                if handle._has_optim_in_backward:
+                    for orig_param in handle.flat_param._params:
+                        # checking grad for None also filters out params
+                        # that don't belong to this rank
+                        if orig_param.grad is not None and hasattr(
+                            orig_param, "_in_backward_optimizers"
+                        ):
+                            for optim in orig_param._in_backward_optimizers:
+                                optim.step()
+                            # Not sure if we need to do this. Setting entire
```

Comment:
In my understanding, we should still do this.

We do not any references keeping the flat gradient alive, and we want the invariant that "if `flat_param.grad is None`, then each original parameter's `.grad is None`".